### PR TITLE
Completed React Effects Exercise-Allyson Greven

### PR DESCRIPTION
**Pull request by allygreven**
effects

### CODE REVIEW

File: react-effects/my-app/src/List.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        setItems(data);
+        setError(null);
+      } catch (error) {
+        console.error('failed to fetch items', error);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    fetchData();
+  });
```
Comment (RobertGardner) [inline react-effects/my-app/src/List.tsx:28]: This must pass a dependency array!

File: react-effects/my-app/src/List.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  //  - When the component mounts:
+  //    - Read the items using `readItems` and update state so the list displays
+  //    - Handle errors from `readItems`
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const data = await readItems();
+        setItems(data);
+        setError(null);
```
Comment (RobertGardner) [inline react-effects/my-app/src/List.tsx:20]: This shouldn't be needed since it's already undefined.

File: react-effects/my-app/src/List.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      } catch (error) {
+        console.error('failed to fetch items', error);
```
Comment (RobertGardner) [inline react-effects/my-app/src/List.tsx:22]: You need to setError here so you can display the message to the user. If you just log it, the user will never learn there was an error.

**Review by RobertGardner — CHANGES_REQUESTED** [review]
Close! And you are the first to submit a solution. By far!

File: react-effects/my-app/src/List.tsx [hunk 4]
```diff
@@ -17,15 +17,16 @@ export function List() {
       try {
         const data = await readItems();
         setItems(data);
-        setError(null);
+        setIsLoading(false);
```
Comment (RobertGardner) [inline react-effects/my-app/src/List.tsx:20]: Since this is called in the finally block you don't need it here.

**Review by RobertGardner — APPROVED** [review]

File: react-effects/my-app/src/List.tsx (hunk 4 above)
Comment (allygreven) [reply react-effects/my-app/src/List.tsx]: ohhhh okay! fixed!

**Pull request merged**